bin/publish

```diff
@@ -1,51 +1,70 @@
 #!/bin/bash
 
+if [[ -z "${TAG_NAME:-}" ]]; then
+  echo "Please supply environment variable TAG_NAME."
+  echo "If you see this error in Jenkins it means the publish script was run"
+  echo "for a build that wasn't triggered by a tag - please check publish stage conditions."
+  exit 1
+fi
+
 set -e
 
 . bin/build_utils
 
 readonly REGISTRY="cyberark"
 readonly VERSION="$(short_version_tag)"
 readonly FULL_VERSION_TAG="$(full_version_tag)"
-readonly INTERNAL_REGISTRY="registry.tld"
 readonly REDHAT_IMAGE="scan.connect.redhat.com/ospid-18d9f51d-9c0c-4031-9f9e-ef08aa2ff409/secretless-broker"
 readonly IMAGES=(
   "secretless-broker"
   "secretless-broker-quickstart"
 )
 
-readonly TAGS=(
-  "$VERSION"
-  "latest"
-)
+function main() {
+  # Ensure TAG_NAME version matches project version
+  if [[ "${TAG_NAME#"v"}" != "${VERSION}" ]]; then
+    echo "TAG_NAME envvar version was '${TAG_NAME#"v"}', expected '${VERSION}'";
+    echo "TAG_NAME envvar version must match project version";
+    exit 1;
+  fi
 
-# fetching tags is required for git_description to work
-git fetch --tags
-git_description=$(git describe)
+  # Accumulate TAGS
+  #
 
-for image_name in "${IMAGES[@]}"; do
-  # always push the tag with the commit hash to internal registry
-  echo "Tagging $INTERNAL_REGISTRY/$image_name:${FULL_VERSION_TAG}"
-  docker tag "$image_name:$FULL_VERSION_TAG" "$INTERNAL_REGISTRY/$image_name:$FULL_VERSION_TAG"
-  echo "Pushing $INTERNAL_REGISTRY/$image_name:${FULL_VERSION_TAG}"
-  docker push "$INTERNAL_REGISTRY/$image_name:$FULL_VERSION_TAG"
+  local TAGS=(
+    "$VERSION"
+  )
 
-  # if it’s not a tagged commit, VERSION will have extra junk (i.e. -g666c4b2), so we won’t publish that commit
-  # only when tag matches the VERSION, push VERSION and latest releases
-  # and x and x.y releases
-  if [ "$git_description" = "v${VERSION}" ]; then
-    echo "Revision $git_description matches version $VERSION exactly. Pushing to Dockerhub..."
+  # Add less specific version tags, eg. given 1.2.3, add 1.2 and 1
+  for tag in $(gen_versions "$VERSION"); do
+    TAGS+=("${tag}")
+  done
+
+  # Add 'latest' tag only when TAG_NAME is the highest semantic tag
+  local LATEST_TAG
+  LATEST_TAG="$(
+    git for-each-ref refs/tags --sort=-refname --format='%(refname:short)' --count=1
```

Comment:
This doesn't do correct ordering since it's lexical:
```
~/checkout/conjur $ git tag v1.10.5
~/checkout/conjur $  git for-each-ref refs/tags --sort=-refname --format='%(refname:short)' 
v1.6.0
v1.5.1
v1.5.0
v1.4.7
v1.4.6
v1.4.4
v1.4.3
v1.4.2
v1.4.1
v1.4.0
v1.3.7
v1.3.6
v1.3.5
v1.3.4
v1.3.3
v1.3.2
v1.3.1
v1.3.0
v1.2.0
v1.10.5 <----
v1.1.2
v1.1.1
v1.1.0
v1.0.1
v1.0.0
v0.9.0
v0.8.1
v0.8.0
v0.7.0
v0.6.0
v0.3.0
v0.2.0
v0.1.1
v0.1.0
delete
1.2.0

Reply:
Good catch. Thanks.`git tag | sort -r --version-sort` does it right.